### ISSUE TITLE
ArrayOutput stops response parsing on empty nested arrays

### ISSUE THREAD

**Issue by TheCycoONE** [lettuce-io/lettuce-core#1327]
#### Current Behavior

When using the redis-cell module v0.2.5 the list of redis commands that Lettuce loads is truncated.

In ReidsCommandFactory.java I put a breakpoint at the return of `getCommands` - from there I can see that commands contains 137 elements when the redis-cell module is loaded, and 204 when it is not loaded.

In redis-cli I run monitor and see the `COMMAND` is used to get the list of commands. When I run that with redis-cell loaded I get 205 entries.

Looking at the list of entries I see the 138th entry is
```
138) 1) "cl.throttle"
     2) (integer) -1
     3) 1) write
     4) (integer) 0
     5) (integer) 0
     6) (integer) 0
     7) (empty array)
```

For some reason Lettuce stops reading the list of commands at that point.

#### Expected behavior/code

The full list of 205 commands should be returned. My custom "CL.THROTTLE" lettuce command should work.

#### Environment

- Lettuce version(s): 5.3.1
- Redis version: 6.0.1
- redis-cell 0.2.5

**Comment by TheCycoONE** [comment]
I knew this was all working 2 years ago, so I've played with older versions. Downgrading redis to 5.0.8 fixes the issue; so it is a redis 6 specific problem.

**Comment by TheCycoONE** [comment]
In redis 5.0.8 the command output is slightly different (missing the empty array element).
```
 30) 1) "cl.throttle"
     2) (integer) -1
     3) 1) write
     4) (integer) 0
     5) (integer) 0
     6) (integer) 0
```

**Comment by mp911de** [comment]
Thanks for reporting the issue. The bug is that `ArrayOutput` doesn't consider nested empty arrays and therefore response processing terminates once the response contains an empty array element.

```
 69) 1) "hget"
     2) (integer) 3
     3) 1) readonly
        2) fast
     4) (integer) 1
     5) (integer) 1
     6) (integer) 1
     7) 1) @read
        2) @hash
        3) @fast
 70) 1) "cl.throttle"
     2) (integer) -1
     3) 1) write
     4) (integer) 1
     5) (integer) 1
     6) (integer) 1
     7) (empty array)
```

**Comment by mp911de** [comment]
That's fixed now in `5.3.2.BUILD-SNAPSHOT`. Care to check whether the snapshot build fixes your issue?

**Comment by TheCycoONE** [comment]
Confirmed working with `5.3.2.BUILD-SNAPSHOT`. Thanks!
